kroxylicious/src/main/java/io/kroxylicious/proxy/micrometer/DefaultMicrometerConfigurationHookContributor.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.micrometer;
+
+import io.kroxylicious.proxy.service.BaseContributor;
+
+public class DefaultMicrometerConfigurationHookContributor extends BaseContributor<MicrometerConfigurationHook> implements MicrometerConfigurationHookContributor {
```

Comment:
I'm not clear why we have `MicrometerConfigurationHookContributor` shouldn't `CommonTagsHook.CommonTagsHookConfig` offer its own Contributor and `StandardBindersHook.StandardBindersHookConfig` its own? I'm not seeing the value of the shared interface.

I could easily be missing something blindingly obvious...

Reply:
> I'm not clear why we have `MicrometerConfigurationHookContributor` shouldn't `CommonTagsHook.CommonTagsHookConfig` offer its own Contributor and `StandardBindersHook.StandardBindersHookConfig` its own? I'm not seeing the value of the shared interface.
> 
> I could easily be missing something blindingly obvious...

I guess what I was really saying is: what does the type field bring us in the model:

```yaml
micrometer:
- type: "StandardBinders"
  config:
    binderNames:
    - "JvmGcMetrics"
```
Compared with

```yaml
micrometer:
  "StandardBinders":
      binderNames:
      - "JvmGcMetrics"
  ```

Reply:
The `type` string is used along with some Jackson resolution magic to obtain the type of the `config` field using the `Contributor`.

Reply:
But yeah I wonder which is preferable:

a) One contributor like:
```
micrometer:
- type: "DefaultConfigHook"
  config:
    binderNames:
    - "JvmGcMetrics"
    commonTags:
       a: "b"
```
b) Using one contributor that offers two types
c) Using two contributor implementations that offer one type each

Reply:
Let's stick as it's currently on this branch.